File: thanhi-shop-app/thanhi-shop-client/src/app/shared/model/Product.ts
import {BaseEntity} from "./BaseEntity";
import {OrderDetail} from "./Order";

/* Product model */
export class Product extends BaseEntity{
  public name?: string;
  public imageUrl?: string;
  public sellPrice?: number = 0;
  public originPrice?: number = 0;
  public statusId?: string;
  public subCateId?: number;
  public hot?: Date;
  public quantity?: number=0;
  public description?: string;
  public sold?: number;
  public orderDetailList?: OrderDetail[];

  constructor( name?: string,
               imageUrl?: string,
               sellPrice?: number,
               originPrice?: number,
               statusId?: string,
               subCateId?: number,
               hot?: Date,
               quantity?: number,
               description?: string,
               sold?: number,
               orderDetailList?: OrderDetail[]) {
    super();
    this.name = name;
    this.imageUrl = imageUrl;
    this.sellPrice = sellPrice;
    this.originPrice = originPrice;
    this.statusId = statusId;
    this.subCateId = subCateId;
    this.hot = hot;
    this.quantity = quantity;
    this.description = description;
    this.sold = sold;
    this.orderDetailList = orderDetailList;
  }
}

File: thanhi-shop-app/thanhi-shop-serversite/src/main/java/thi/app/model/entity/Product.java
package thi.app.model.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;
import org.hibernate.search.mapper.pojo.mapping.definition.annotation.FullTextField;
import org.hibernate.search.mapper.pojo.mapping.definition.annotation.Indexed;

import javax.persistence.*;
import javax.validation.constraints.Min;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;
import java.io.Serializable;
import java.math.BigDecimal;
import java.time.Instant;
import java.util.Set;


@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
@Table(name = "tb_products")
@Indexed
public class Product extends AbstractEntity implements Serializable {

    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull
    @Size(max = 255)
    @Column(length = 255)
    @FullTextField
    private String name;

    @Size(max = 256)
    @Column(length = 256)
    private String imageUrl;

    @Min(1)
    @Column
    private BigDecimal sellPrice;

    @Min(1)
    @Column
    private BigDecimal originPrice;
    @ManyToOne
    @JoinColumn(name = "status_id")
    private Status status;

    @ManyToOne
    @JoinColumn(name = "sub_cate_id")
    private SubCategory subCategory;

    @Column
    private Instant hot = Instant.now();

    @Column
    @Min(0)
    private Integer quantity;

    @Size(min = 1,max = 500)
    @Column(length = 500)
    private String description;

    @Min(0)
    @Column
    private Integer sold;

    @OneToMany(mappedBy = "product")
    @JsonIgnore
    private Set<OrderDetail> orderDetailList;
}
